pkg/kubectl/rolling_updater.go

```diff
@@ -208,13 +208,20 @@ func (r *RollingUpdater) Update(config *RollingUpdaterConfig) error {
 	if err != nil {
 		return err
 	}
-	// Further validation.
-	if maxUnavailable == 0 && maxSurge == 0 {
+	// Validate maximums.
+	if desired > 0 && maxUnavailable == 0 && maxSurge == 0 {
 		return fmt.Errorf("one of maxSurge or maxUnavailable must be specified")
 	}
 	// The minumum pods which must remain available througout the update
 	// calculated for internal convenience.
-	minAvailable := original - maxUnavailable
+	minAvailable := desired - maxUnavailable
```

Comment:
(I assume the intent is to guard against negative values; if min is used, then minAvailable would always be 0 or negative)